File: backend-java/patient-service/src/main/java/com/cdm/patient/service/PatientService.java
package com.cdm.patient.service;

import com.cdm.common.security.IdentityPayload;
import com.cdm.patient.entity.PatientProfileEntity;
import com.cdm.patient.repository.PatientRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PatientService {
    private final PatientRepository repository;

    public PatientService(PatientRepository repository) {
        this.repository = repository;
    }

    public List<PatientProfileEntity> findAll(IdentityPayload identity) {
        return repository.findAllByContext(identity.getTenantId(), identity.getAllowedOrgIds());
    }

    public PatientProfileEntity createPatient(IdentityPayload identity, String name, String gender) {
        PatientProfileEntity entity = new PatientProfileEntity();
        entity.setId(com.cdm.common.util.SnowflakeIdGenerator.nextId());
        entity.setTenantId(identity.getTenantId());
        entity.setOrgId(identity.getOrgId());
        entity.setName(name);
        entity.setGender(gender);
        return repository.save(entity);
    }
}

File: backend-nestjs/apps/patient/src/patient/patient.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { In, Repository } from 'typeorm';
import { PatientEntity } from './patient.entity.js';
import { IdentityPayload, nextId } from '@cdm/shared';
import type { PatientVO } from '@cdm/shared';

@Injectable()
export class PatientService {
  constructor(
    @InjectRepository(PatientEntity)
    private readonly repo: Repository<PatientEntity>,
  ) {}

  async findAll(identity: IdentityPayload): Promise<PatientVO[]> {
    const entities = await this.repo.find({
      where: {
        tenantId: identity.tenantId,
        orgId: In(identity.allowedOrgIds),
      },
    });
    return entities.map(PatientService.toVO);
  }

  async createPatient(identity: IdentityPayload, data: { name: string; gender: string }): Promise<PatientVO> {
    const entity = this.repo.create({
      id: nextId(),
      tenantId: identity.tenantId,
      orgId: identity.orgId,
      name: data.name,
      gender: data.gender,
    });
    const saved = await this.repo.save(entity);
    return PatientService.toVO(saved);
  }

  static toVO(entity: PatientEntity): PatientVO {
    return {
      id: entity.id,
      tenantId: entity.tenantId,
      orgId: entity.orgId,
      name: entity.name,
      gender: entity.gender,
    };
  }
}
